lib/commands/gesture.js

```diff
@@ -0,0 +1,43 @@
+import logger from '../logger';
+
+let commands = {};
+
+/**
+ * Perform touch command
+ */
+commands.performTouch = async function (gestures) {
+  //Convert and pass on the touch command
+  if (gestures.length === 1 && gestures[0].action === 'tap') {
+    return await this.handleTap(gestures[0]);
+  }
+  //TODO: Handle multiple gestures, or anything else but tap
+};
+
+commands.handleTap = async function (gesture) {
+  let options = gesture.options;
+
+  let nativeDriver = await this.getNativeDriver();
+  let gestureObject = {};
+  if (options.x) {
+    gestureObject.x = options.x;
+  }
+  if (options.y) {
+    gestureObject.y = options.y;
+  }
+  let endpoint = 'tap';
+  //Either an element or if there is none given, the WebDriverAgent API wants a param here
+  if (options.element) {
+    endpoint += '/' + options.element;
+  } else {
+    endpoint += '/0';
+  }
+  nativeDriver.sendCommandWithSession(endpoint, gestureObject, 'POST');
```

Comment:
So, we really like abstracting all specifics of the JSON wire protocol out of the internal driver code.

I kind of see why this might be needed though.

Can the code be changed in the facebook webdriveragent project? They should be accepting the full webdriver api, so our methods don't need to do anything besides proxy commands.

Also, be aware that the Selenium spec might have a more updated way of handling touch gestures. We should build to the latest version of that spec.